keps/sig-api-machinery/1886-jsonpath/README.md

```diff
@@ -0,0 +1,333 @@
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+- [ ] Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+- [ ] KEP approvers have approved the KEP status as `implementable`
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+[original JSONPath syntax]: https://goessner.net/articles/JsonPath
+[existing JSONPath doc]: https://github.com/kubernetes/website/blob/ecc27bbbe70f92d031fa52be76bb9471b2e83152/content/en/docs/reference/kubectl/jsonpath.md
+[JSON Strategic Merge Patch]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md
+[go template]: https://golang.org/pkg/text/template/
+## Summary
+
+This proposal aims to clarify the JSONPath template syntax supported by Kubernetes
+and make some enhancement to meet today's user needs.
+
+In addition to [original JSONPath syntax], this JSONPath enhancement mainly allow users to apply more complex, flexible filter
+on JSON-formatted resources using kubectl.
+Enables the ability to extract desired fields with a single and simple statement.
+
+## Motivation
+
+There are already many issues/pull-requests about JSONPath in Kubernetes.
+However, none of them were approved for lacking standard.
+Besides, the [existing JSONPath doc] for Kubernetes is ambiguous:
+No supported logical operator list, either no clear syntax.
+
+In fact, the most authoritative JSONPath standard was proposed in 2007 and never been updated in past 13 years. See [original JSONPath syntax].
+It's unreasonable and self deceiving to reject evolving for better user experience just because of an outdated standard.
+Like [JSON Strategic Merge Patch], We need someone step out for the first change.
+
+The job for this KEP is basically in 2 parts:
+- Clarify supported syntax
+- Enhancement
+
+Both parts should have clear and sufficient documentation about syntax/examples.
+
+### Goals
+
+* Document relationship with [original JSONPath syntax]
+* Have a clear list of supported JSONPath syntax, including original and enhanced syntax
```

Comment:
I presume that “provide a documented, widely accepted definition of JSONpath” (in general, not just for Kubernetes) is a non-goal?

Maybe maybe that explicit?

Reply:
I agree. In fact, I am considering writing a go-jsonpath package, which is intended for entire golang community.
For now I haven't found such a package with completed documentation and any enhancement.
Listing it into non-Goal section sounds a good idea.

Reply:
That sounds good to me, actually-- if you have an existing well-documented library and propose that we adopt it, that might be an easier conversation than the one about defining a superset we want to support.

Reply:
That's great. Prepare working for it.